app/js/instances/templates/instance.html

```diff
@@ -69,11 +69,14 @@
   </div>
 
   <div class="row">
-    <div class="col-md-12" ng-repeat="(name, progress) in loadingImages">
-      <span>Pulling <strong>{{ name }}</strong></span>
-      <div class="progress">
-        <div class="progress-bar progress-bar-info" role="progressbar" aria-valuemin="0" aria-valuemax="100" ng-style="{'width': progress}"></div>
-      </div>
+    <div class="col-md-12" ng-repeat="(name, layers) in loadingImages track by name">
+      <span>Pulling <strong>{{ name }}</strong> <i class="fa fa-spinner fa-spin"></i></span>
+      <div ng-repeat="layer in layers track by layer.id">
+        <div class="progress">
+          <div class=" progress-bar progress-bar-info" role="progressbar" aria-valuemin="0" aria-valuemax="100" ng-style="{'width': layer.progress}">
```

Comment:
From a UX perspective, the centered progress bar text may get a little annoying as the bar grows. Could we push it off to the top or left of the progress bar?